packages/flutter_tools/test/integration.shard/isolated/native_assets_test_utils.dart

```diff
@@ -65,18 +67,30 @@ Future<void> addLinkHookDependency(String packageName, Directory packageDirector
       .childDirectory('link_hook');
   expect(linkHookDirectory, exists);
 
-  final File pubspecFile = packageDirectory.childFile('pubspec.yaml');
-  final String pubspecOld =
-      (await pubspecFile.readAsString()).replaceAll('\r\n', '\n');
-  final String pubspecNew = pubspecOld.replaceFirst('''
-dependencies:
-''', '''
-dependencies:
-  link_hook:
-    path: ${linkHookDirectory.path}
-''');
-  expect(pubspecNew, isNot(pubspecOld));
-  await pubspecFile.writeAsString(pubspecNew);
+  final File linkHookPubspecFile = linkHookDirectory.childFile('pubspec.yaml');
+  final File thisPubspecFile = packageDirectory.childFile('pubspec.yaml');
+
+  final Map<String, Object?> linkHookPubspec = _pubspecAsMutableJson(linkHookPubspecFile .readAsStringSync());
```

Comment:
nit: there is a rogue whitespace before `.`.

Reply:
Done.